### PR TITLE
Externalize issuer's dPoPJwtFactory

### LINKED ISSUE #284: Externalize Issuer's internal DPoPJwtFactory
Expose `Issuer` component's internal `dPoPJwtFactory` constructed based on the cababilities of the authorization server selected to authorize the issuance. It can be utilized to other calls based on the DPoP Access Token issued from authorization server.
Also DPoPJwtFactory class implements functinality of [Demonstrating Proof of Possession](https://www.rfc-editor.org/rfc/rfc9449.html). Can be exposed as a unitlity

### PR DESCRIPTION
**Pull request by vafeini**
Externalize issuer's dPoPJwtFactory constructed under the hood when initializing an issuer component

closes #284

### CODE REVIEW

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/Types.kt [hunk 1]
```diff
@@ -356,3 +356,7 @@ value class CoseCurve private constructor(val value: Int) {
         }
     }
 }
+
+enum class Htm {
```
Comment (babisRoutis) [inline src/main/kotlin/eu/europa/ec/eudi/openid4vci/Types.kt:360]: Better keep those to DPoP file

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/DPoP.kt [hunk 2]
```diff
@@ -31,14 +30,12 @@ import java.util.*
 import com.nimbusds.oauth2.sdk.dpop.DPoPProofFactory as NimbusDPoPProofFactory
 import com.nimbusds.oauth2.sdk.token.DPoPAccessToken as NimbusDPoPAccessToken
 
-internal enum class Htm {
-    GET, HEAD, POST, PUT, DELETE, CONNECT, OPTIONS, TRACE
-}
+const val DPoP = "DPoP"
```
Comment (babisRoutis) [inline src/main/kotlin/eu/europa/ec/eudi/openid4vci/DPoP.kt:33]: Does this have to be public?

**Review by babisRoutis — DISMISSED** [review]
@vafeini  Just a couple minor comment, for your consideration.

Also, please update PR or even better add a issue providing the intention for making this public.

**Review by babisRoutis — DISMISSED** [review]

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/Types.kt (hunk 1 above)
Comment (vafeini) [reply src/main/kotlin/eu/europa/ec/eudi/openid4vci/Types.kt]: Done

**Comment by babisRoutis** [comment]
@vafeini Check your build please

File: src/main/kotlin/eu/europa/ec/eudi/openid4vci/DPoP.kt (hunk 2 above)
Comment (vafeini) [reply src/main/kotlin/eu/europa/ec/eudi/openid4vci/DPoP.kt]: Marked as private

**Pull request merged**